synchronicity/synchronizer.py

```diff
@@ -152,11 +159,22 @@ def _start_loop(self):
             is_ready = threading.Event()
 
             def thread_inner():
+                async def loop_delay_monitor():
+                    while 1:
+                        t0 = time.monotonic()
+                        await asyncio.sleep(self.loop_delay_monitor_period)
+                        duration = time.monotonic() - t0
+                        delay = duration - self.loop_delay_monitor_period
```

Comment:
Could our estimate of the true time the event loop was blocked be off by up to `loop_delay_monitor_period`? In other words, is the right way to think about it that another task could start blocking immediately after we sleep, or immediately before we try to wake, and we wouldn't be able to tell the difference the way we're calculating / reporting the delay here?

(This is mostly a question for my own understanding — it's probably not critical to report metrics exactly here).

Reply:
Yes, technically the event loop could have been blocked for up to `loop_delay_monitor_period` while this coroutine is sleeping and we wouldn't detect it unless the blockage overlapped with our expected return time. So worst case we could miss event loop blockage by `loop_delay_monitor_period + loop_delay_monitor_threshold` seconds (since we don't report unless we pass the threshold either).

But for Modal's purposes the most critical event loop blockage would be anything in the 10+ second magnitude (where heartbeats start to time out), so this should be fine (and we can adjust the period and threshold)

Reply:
If we want to be pedantic we could say “by up to” and give the upper bound but agree a second margin of error isn’t a huge deal.